ndcube/utils/wcs.py

```diff
@@ -429,3 +429,36 @@ def array_indices_for_world_objects(wcs, axes=None):
         array_index = convert_between_array_and_pixel_axes(pixel_index, wcs.pixel_n_dim)
         array_indices[oinds] = tuple(array_index[::-1])  # Invert from pixel order to array order
     return tuple(ai for ai in array_indices if ai)
+
+
+def validate_wcs(source_wcs, target_wcs):
+    """
+    Checks if two WCS objects are comptible with each other for reprojecting an NDCube on another.
+
+    Parameters
+    ----------
+    source_wcs : `astropy.wcs.WCS` or any other object that implements
+                    `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
```

Comment:
I would simplify both of these to just this:
```suggestion
    source_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS` or `astropy.wcs.wcsapi.BaseLowLevelWCS`
```